nixos/modules/services/editors/emacs.nix

```diff
@@ -85,7 +99,10 @@ in
 
     environment.systemPackages = [ cfg.package editorScript ];
 
-    environment.variables.EDITOR = mkIf cfg.defaultEditor (mkOverride 900 "emacseditor");
+    environment.variables = {
+      EDITOR = mkIf cfg.defaultEditor (mkOverride 900 "emacseditor");
+      ALTERNATE_EDITOR = cfg.alternateEditor;
```

Comment:
There is an issue of using a environment variable `ALTERNATE_EDITOR` instead of setting `--alternate-editor` in the script.  If a user changes`alternateEditor` and runs `nixos-rebuild test (or switch)`, `ALTERNATE_EDITOR` is not updated in the current shell.

Reply:
Welp, that's an issue with all envvars, no? I would imagine an Emacs user rather test the alteditor with `emacsclient -a "new value"` or, at worst, with `ALTERNATE_EDITOR="new value" emacsclient`.

(this assumes that a user did read the option's description)

Reply:
> that's an issue with all envvars, no?

Yes, and that issue can be avoided here if we do not use env var in the first place.

So what is the benefit of not using `-a` in the script?